### PR TITLE
INFINITY-2429 Fix Go vendoring and use vendoring consistently

### PR DESCRIPTION
**Pull request by nickbp**
Go is confused by directories that reference other directories in the same repo, particularly when external repos are being pulled in as well. Adds a shared `govendor` dir to be symlinked by CLIs/bootstrap in dcos-commons. Replaces O(Nframeworks) vendor copies that are currently present. External repos wouldn't need to jump through these hoops.

Removes a `go get` command from `build_go_exe.sh`, as all libraries should be included in the `govendor` directory anyway. And this effectively enforces that libraries are vendored to avoid spontaneous build-time issues.

### CODE REVIEW

File: clivendor/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# CLI Vendor hack
+
+This is a workaround for the go tooling being confused by `frameworks/myframework/cli/` pointing into `cli/` when using vendoring.
+
+Frameworks in dcos-commons/frameworks/ should create a 'vendor' symlink which points to here. In other words, a symlink at `frameworks/yoursvc/cli/dcos-yoursvc/vendor` should point to `clivendor/`.
```
Comment (elezar) [inline clivendor/README.md]: Should we add this to the `new-framework.sh` script?

File: clivendor/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# CLI Vendor hack
+
+This is a workaround for the go tooling being confused by `frameworks/myframework/cli/` pointing into `cli/` when using vendoring.
```
Comment (elezar) [inline clivendor/README.md]: replace: "go" -> "Go"?

File: clivendor/vendor.json [hunk 3]
```diff
@@ -21,6 +21,18 @@
...
 		},
 		{
+			"checksumSHA1": "ulFwBsxjpyu+bk44+LKDl8AKQiA=",
+			"path": "github.com/aryann/difflib",
+			"revision": "e206f873d14a916d3d26c40ab667bca123f365a3",
+			"revisionTime": "2017-07-10T04:42:30Z"
+		},
+		{
+			"checksumSHA1": "AIxWUvzOjHni8IwkElJw+b1IES8=",
+			"path": "github.com/nickbp/mustache",
```
Comment (elezar) [inline clivendor/vendor.json]: Do the CLIs actually use `mustache` or `difflib`? Or is this vendor directory now more general than just CLI?

File: clivendor/README.md (hunk 1 above)
Comment (nickbp) [reply clivendor/README.md]: Open question: delete that script as of this PR?

File: clivendor/vendor.json (hunk 3 above)
Comment (nickbp) [reply clivendor/vendor.json]: This is the union of both CLI and bootstrap dependencies

Comment (elezar) [reply clivendor/vendor.json]: Should we then change the folder name `clivendor` to `vendor` or `go-vendor`?

File: clivendor/README.md (hunk 1 above)
Comment (elezar) [reply clivendor/README.md]: What is the documented way to add a new framework and/or create a new standalone service. But yes, the scripts that are not relevant anymore should be removed.

File: clivendor/vendor.json (hunk 3 above)
Comment (nickbp) [reply clivendor/vendor.json]: I now realize that the name `clivendor` isn't quite accurate, I'll update this to be `govendor`, even though that's technically also the name of the tool used to manage vendoring.

Comment (nickbp) [reply clivendor/vendor.json]: Actually I'll try `govendor` just to follow the example of e.g. `frameworks/helloworld` of avoiding punctuation

**Comment by paegun** [comment]
infinity/hdfs build failure `test_sanity.test_modify_app_config` highly relates to https://github.com/mesosphere/dcos-commons/pull/1619

**Comment by elezar** [comment]
Just looked at https://github.com/mesosphere/dcos-commons/blob/a39ce51bdff6d8e3ebf66968e7d13f078306c773/cli/README.md and it seems `govendor` is a go package/utility.

**Comment by nickbp** [comment]
Missing approval

**Pull request merged**